### PR TITLE
Update CDF reference

### PR DESCRIPTION
**Pull request by Jesus89**
Closes https://github.com/CartoDB/cartoframes/issues/1307

### CODE REVIEW

File: cartoframes/viz/layer.py [hunk 1]
```diff
@@ -19,8 +19,8 @@ class Layer(object):
     notebook to get a preview of a Layer.
 
     Args:
-        source (str, :py:class:`CartoDataFrame <cartoframes.CartoDataFrame>`,
-            pandas.DataFrame, geopandas.GeoDataFrame): The source data.
+        source (str, pandas.DataFrame, geopandas.GeoDataFrame,
+            :py:class:`CartoDataFrame <cartoframes.CartoDataFrame>`): The source data.
```
Comment (andy-esch) [inline cartoframes/viz/layer.py]: How about explicitly saying "table, query, or three dataframes listed". Some people might not know what `str` could give them

File: cartoframes/core/cartodataframe.py [hunk 2]
```diff
@@ -33,29 +33,61 @@ def from_carto(*args, **kwargs):
         """
         Alternate constructor to create a CartoDataFrame from a table or SQL query in CARTO.
         It is needed to set up the :py:class:`cartoframes.auth.Credentials`.
-        Equivalent to :py:meth:`read_carto <cartoframes.io.read_carto>`.
+        Equivalent to :py:meth:`cartoframes.read_carto <cartoframes.io.carto.read_carto>`.
+
+        Args:
```
[the remark this answers is not in the document]
Comment (Jesus89) [reply cartoframes/core/cartodataframe.py]: We were using *args, **kwargs before the doc, but now it makes total sense to add explicit params now.

File: cartoframes/core/cartodataframe.py [hunk 3]
```diff
@@ -65,7 +97,7 @@ def from_carto(*args, **kwargs):
     @classmethod
     def from_file(cls, filename, **kwargs):
         """
-        Alternate constructor to create a CartoDataFrame from a file.
+        Alternate constructor to create a CartoDataFrame from a Shapefile or GeoJSON file.
```
[the remark this answers is not in the document]
Comment (Jesus89) [reply cartoframes/core/cartodataframe.py]: I don't think CSV is available in geopandas `from_file`. We can implement this feature using internally pd.read_csv, but does it make sense in order to achieve CDF.from_file('data.csv')?

File: cartoframes/core/cartodataframe.py [hunk 4]
```diff
@@ -102,15 +143,36 @@ def to_carto(self, *args, **kwargs):
         """
         Upload a CartoDataFrame to CARTO. It is needed to set up the
         :py:class:`cartoframes.auth.Credentials`.
-        Equivalent to :py:meth:`to_carto <cartoframes.io.to_carto>`.
+        Equivalent to :py:meth:`cartoframes.to_carto <cartoframes.io.carto.to_carto>`.
+
+        Args:
```
[the remark this answers is not in the document]
Comment (Jesus89) [reply cartoframes/core/cartodataframe.py]: :+1:

File: cartoframes/core/cartodataframe.py [hunk 5]
```diff
@@ -120,22 +182,92 @@ def to_carto(self, *args, **kwargs):
 
     def viz(self, *args, **kwargs):
         """
-        Creates a :py:class:`Map <cartoframes.viz.Map>` visualization
+        Creates a quick :py:class:`Map <cartoframes.viz.Map>` visualization. The parameters
+        are passed directly to the Layer (style, popup, legend, widgets, etc.).
+
```
[the remark this answers is not in the document]
Comment (Jesus89) [reply cartoframes/core/cartodataframe.py]: :+1:

File: cartoframes/viz/layer.py (hunk 1 above)
Comment (Jesus89) [reply cartoframes/viz/layer.py]: OK

Comment (Jesus89) [reply cartoframes/viz/layer.py]: Since we are using this (...) section for the types, I would improve the definition of the parameter instead.

**Comment by Jesus89** [comment]
Suggestions added. Please take another review when you have time :)

File: cartoframes/core/cartodataframe.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (andy-esch) [reply cartoframes/core/cartodataframe.py]: I don't think it makes sense to do it. I like inheriting the behavior of geopandas so users have clear expectations.

My personal experience with `GeoDataFrame.from_file` is that it is surprisingly difficult or unintuitive to use

[the remark this answers is not in the document]
Comment (andy-esch) [reply cartoframes/core/cartodataframe.py]: I vote for linking to the same docs that geopandas does so users have a solid understanding of what does or does not work in this method

[the remark this answers is not in the document]
Comment (Jesus89) [reply cartoframes/core/cartodataframe.py]: Aha. So this is a good point to make clear in the doc that only allows Shapefile and GeoJSON, afaik.

[the remark this answers is not in the document]
Comment (Jesus89) [reply cartoframes/core/cartodataframe.py]: I will add a link in our docstring.

[the remark this answers is not in the document]
Comment (Jesus89) [reply cartoframes/core/cartodataframe.py]: Done.

**Pull request merged**